packages/agreement-process/src/services/agreementService.ts

```diff
@@ -391,17 +396,40 @@ export function agreementServiceBuilder(
         readModelService
       );
 
-      const nextState = nextStateByAttributes(
+      const nextStateByAttributes = nextStateByAttributesFSM(
         agreement.data,
         descriptor,
         consumer
       );
 
+      const suspendedByPlatform = suspendedByPlatformFlag(
+        nextStateByAttributes
+      );
+
+      const setToMissingCertifiedAttributesByPlatformEvent =
+        maybeCreateSetToMissingCertifiedAttributesByPlatformEvent(
+          agreement,
+          nextStateByAttributes,
+          suspendedByPlatform,
+          correlationId
+        );
+
+      if (setToMissingCertifiedAttributesByPlatformEvent) {
+        /* In this case, it means that one of the certified attributes is not
+          valid anymore. We put the agreement in the missingCertifiedAttributes state
+          and fail the submission */
+
+        await repository.createEvent(
+          setToMissingCertifiedAttributesByPlatformEvent
+        );
+        throw agreementSubmissionFailed(agreement.data.id);
+      }
```

Comment:
could be a separated method like assertion or validator ?

Reply:
Being this specific and unusual case where we save and event in the event store and throw an error, I would rather leave it explicit - also, in the two cases where this happens, it throws different errors: `agreementSubmissionFailed` in one case, `agreementActivationFailed` in the other

Reply:
Make sense